### PR TITLE
[PRM-2332] Moved TTP to Estimate block

### PR DESCRIPTION
**Pull request by JBlacklaws**

### CODE REVIEW

File: app/views/CalculationLPP2View.scala.html [hunk 1]
```diff
@@ -87,13 +87,12 @@ <h1 class="govuk-heading-l govuk-!-margin-bottom-5">@messages("calculation.headi
...
-  @if(isTTPActive) {
-    @insetText(InsetText(
-      id = Some("ttp-inset-text"),
-      content = HtmlContent(getMessage("calculation.ttp.insetText"))))
-  }
-
   @if(isEstimate) {
+    @if(isTTPActive) {
+        @insetText(InsetText(
+            id = Some("ttp-inset-text"),
```
Comment (LewisMitchell) [inline app/views/CalculationLPP2View.scala.html:93]: Just thinking if you want to add a class to this so you can check if _any_ TTP content exists?

File: app/views/CalculationLPP2View.scala.html [hunk 2]
```diff
@@ -87,13 +87,12 @@ <h1 class="govuk-heading-l govuk-!-margin-bottom-5">@messages("calculation.headi
...
+            content = HtmlContent(getMessage("calculation.ttp.insetText"))))
+    }
     <h2 class="govuk-heading-s">@messages("calculation.h2.estimates")</h2>
     @if(isTTPActive) {
       <p class="govuk-body">@getMessage("calculation.p.3.estimates.lpp.ttpActive")</p>
```
Comment (LewisMitchell) [inline app/views/CalculationLPP2View.scala.html:98]: Just thinking if you want to add a class to this so you can check if _any_ TTP content exists?

File: it/controllers/CalculationControllerISpec.scala [hunk 3]
```diff
@@ -615,6 +615,7 @@ class CalculationControllerISpec extends IntegrationSpecCommonBase with FeatureS
       parsedBody.select("#main-content .govuk-summary-list__row").get(1).select("dd").text() shouldBe "£0.00"
       parsedBody.select("#main-content .govuk-summary-list__row").get(2).select("dt").text() shouldBe "Left to pay"
       parsedBody.select("#main-content .govuk-summary-list__row").get(2).select("dd").text() shouldBe "£123.45"
+      parsedBody.select("#ttp-inset-text").isEmpty shouldBe true
```
Comment (LewisMitchell) [inline it/controllers/CalculationControllerISpec.scala:618]: If you add the class I mentioned, you can then do 
```suggestion
      parsedBody.select(".whatever-class").isEmpty shouldBe true
```

File: test/views/CalculationViewSpec.scala [hunk 4]
```diff
@@ -107,6 +108,10 @@ class CalculationViewSpec extends SpecBase with ViewBehaviours with ViewUtils {
           doc.select(Selector.breadcrumbWithLink(2)).attr("href") shouldBe appConfig.vatOverviewUrl
           doc.select(Selector.breadcrumbWithLink(3)).attr("href") shouldBe controllers.routes.IndexController.onPageLoad.url
         }
+
+        "not display TPP content" in {
```
Comment (LewisMitchell) [inline test/views/CalculationViewSpec.scala:112]:
```suggestion
        "not display TTP content" in {
```

File: test/views/CalculationViewSpec.scala [hunk 5]
```diff
@@ -107,6 +108,10 @@ class CalculationViewSpec extends SpecBase with ViewBehaviours with ViewUtils {
...
+          doc.select("#ttp-inset-text").isEmpty shouldBe true
```
Comment (LewisMitchell) [inline test/views/CalculationViewSpec.scala:113]: If you add the class I mentioned, you can then do 
```suggestion
          doc.select(".whatever-class").isEmpty shouldBe true
```

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: app/views/CalculationLPP2View.scala.html (hunk 1 above)
Comment (JBlacklaws) [reply app/views/CalculationLPP2View.scala.html:93]: Inset text already has a class by default "govuk-inset-text" but I could change the id to be "ttp-content", then add that to the other tpp line below?

Comment (LewisMitchell) [reply app/views/CalculationLPP2View.scala.html:93]: My thoughts were to have 'ttp-info' as a class on both the inset text and the TTP text below. That means in the test you can check if _all_ of the TTP content on the page exists or not e.g.,
parsedBody.select(".whatever-class").isEmpty shouldBe true

File: test/views/CalculationViewSpec.scala [hunk 6]
```diff
@@ -107,6 +108,10 @@ class CalculationViewSpec extends SpecBase with ViewBehaviours with ViewUtils {
           doc.select(Selector.breadcrumbWithLink(2)).attr("href") shouldBe appConfig.vatOverviewUrl
           doc.select(Selector.breadcrumbWithLink(3)).attr("href") shouldBe controllers.routes.IndexController.onPageLoad.url
         }
+
+        "not display TTP content" in {
+          doc.select(".ttp-content").isEmpty shouldBe true
+        }
```
Comment (LewisMitchell) [inline test/views/CalculationViewSpec.scala:114]: Wondering if it's worth adding this test to where the penalty is _not_ estimated?

**Review by LewisMitchell — CHANGES_REQUESTED** [review]

File: test/views/CalculationViewSpec.scala [hunk 7]
```diff
@@ -192,6 +197,47 @@ class CalculationViewSpec extends SpecBase with ViewBehaviours with ViewUtils {
         )
 
         behave like pageWithExpectedMessages(expectedContent)
+
+        "not display TTP content" in {
+          doc.select(".ttp-content").isEmpty shouldBe true
+        }
```
Comment (LewisMitchell) [inline test/views/CalculationViewSpec.scala:203]:
```suggestion
        "not display TTP content" in {
          doc.select(".ttp-content").isEmpty shouldBe true
        }
```

File: test/views/CalculationViewSpec.scala [hunk 8]
```diff
@@ -192,6 +197,47 @@ class CalculationViewSpec extends SpecBase with ViewBehaviours with ViewUtils {
...
+          Selector.summaryListRowKey(2) -> th2LPP,
+          Selector.summaryListRowValue(2) -> "£50.50",
+          Selector.summaryListRowKey(3) -> th3LPP,
+          Selector.summaryListRowValue(3) -> "£40.10",
+          Selector.summaryListRowKey(4) -> th4LPP,
+          Selector.summaryListRowValue(4) -> "£10.40",
+          Selector.link -> link
+        )
+
+        behave like pageWithExpectedMessages(expectedContent)
```
Comment (LewisMitchell) [inline test/views/CalculationViewSpec.scala:240]: Wonder if it's worth adding the TTP not displayed test here?

**Review by LewisMitchell — APPROVED** [review]

**Pull request merged**